arelle/ModelValue.py

```diff
@@ -484,124 +606,185 @@ def time(value, castException=None):
     return Time(int(match.group(1)),int(match.group(2)),int(match.group(3)),ms,tzinfo(match.group(5)))
 
 class Time(datetime.time):
-    def __new__(cls, hour=0, minute=0, second=0, microsecond=0, tzinfo=None):
+
+    hour24: bool
+
+    def __new__(cls, hour: int | datetime.time = 0, minute: int = 0, second: int = 0, microsecond: int = 0, tzinfo: datetime.timezone | None = None) -> Time:
```

Comment:
The hour type can be changed once the max/min stuff is taken care of/ignored, but I also noticed the tzinfo type is wrong:
```suggestion
    def __new__(cls, hour: int = 0, minute: int = 0, second: int = 0, microsecond: int = 0, tzinfo: datetime.tzinfo | None = None) -> Time:
```